### PR TITLE
Build calico/node image from AlmaLinux

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

CentOS Stream 8 was EoL'ed on 31 May, 2024. This change switches build base to AlmaLinux which is binary compatible with RHEL/UBI.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/.gitignore [hunk 1]
```diff
@@ -20,9 +20,12 @@
...
 .go-pkg-cache/
 /artifacts/
-/bpf-gpl/include/libbpf/src/amd64/
-/bpf-gpl/include/libbpf/src/arm64/
+bpf-gpl/include/libbpf/src/amd64/
+bpf-gpl/include/libbpf/src/arm64/
 bpf-gpl/include/libbpf/src/ppc64le/
+bpf-gpl/include/libbpf/src/s390x/
+bpf-gpl/*.ll.tmp
+bpf-gpl/ut/*.ll.tmp
```
Comment (hjiawei) [inline felix/.gitignore:28]: Ignore temporary bpf files during the build.

File: node/Dockerfile.amd64 [hunk 2]
```diff
@@ -91,30 +91,21 @@ ARG RUNIT_VER
...
-# Since the ubi repos do not contain all the packages we need (they're missing conntrack-tools),
-# we're using CentOS repos for all our packages. Using packages from a single source (CentOS) makes
-# it less likely we'll run into package dependency version mismatches.
-#
-# NOTE: new packages need to be added to the keep-list in clean-up-filesystem.sh.
-COPY centos.repo /etc/yum.repos.d/
-RUN rm /etc/yum.repos.d/ubi.repo && \
-    touch /in-the-container && \
-    microdnf install \
+RUN microdnf install \
```
Comment (hjiawei) [inline node/Dockerfile.amd64:104]: Download as many packages as possible from UBI.

File: node/Dockerfile.amd64 [hunk 3]
```diff
@@ -125,18 +116,24 @@ RUN rm /etc/yum.repos.d/ubi.repo && \
...
+
+# Since the ubi repos do not contain all the packages we need (they're missing conntrack-tools),
+# we're using AlmaLinux repos for missing packages.
+COPY almalinux.repo /etc/yum.repos.d/almalinux.repo
+
+RUN microdnf --enablerepo=baseos install \
+    iproute-tc \
+    # Needed for conntrack
+    libnetfilter_cthelper libnetfilter_cttimeout libnetfilter_queue \
+    conntrack-tools
```
Comment (hjiawei) [inline node/Dockerfile.amd64:132]: AlmaLinux is binary compatible with RHEL so install missing packages from it.

File: node/clean-up-filesystem.sh [hunk 4]
```diff
@@ -255,6 +255,7 @@ packages_to_keep=(
   langpacks
   libacl
   libattr
+  libbpf
```
Comment (sridhartigera) [inline node/clean-up-filesystem.sh:258]: Is this needed?

Comment (hjiawei) [reply node/clean-up-filesystem.sh:258]: The cleanup script complaints about libbpf.so.0.5.0 missing after switching to AlmaLinux base so I added it to the keep list.

**Review by sridhartigera — APPROVED** [review]
LGTM

**Pull request merged**